addons/account/wizard/account_automatic_entry_wizard.py

```diff
@@ -0,0 +1,387 @@
+# -*- coding: utf-8 -*-
+from odoo import api, fields, models, _
+from odoo.exceptions import UserError
+from odoo.tools.misc import format_date, formatLang
+
+from collections import defaultdict
+import json
+
+class AutomaticEntryWizard(models.TransientModel):
+    _name = 'account.automatic.entry.wizard'
+    _description = 'Create Automatic Entries'
+
+    # General
+    action = fields.Selection([('accrual', 'Change Period'), ('transfer', 'Change Account')], required=True)
+    data = fields.Text(compute="_compute_data")
+    preview_data = fields.Text(compute="_compute_preview_data")
+    move_line_ids = fields.Many2many('account.move.line')
+    date = fields.Date(required=True, default=fields.Date.today())
+    company_id = fields.Many2one('res.company', required=True, readonly=True)
+    company_currency_id = fields.Many2one('res.currency', related='company_id.currency_id')
+    percentage = fields.Float("Percentage", compute='_compute_percentage', readonly=False, store=True)
+    total_amount = fields.Monetary(compute='_compute_total_amount', store=True, readonly=False, currency_field='company_currency_id')
+    journal_id = fields.Many2one('account.journal', required=True, readonly=False, string="Journal",
+        domain="[('company_id', '=', company_id), ('type', '=', 'general')]",
+        related="company_id.automatic_entry_default_journal",
+        help="Journal where to create the entry.")
+
+    # Accrual
+    account_type = fields.Selection([('income', 'Revenue'), ('expense', 'Expense')], compute='_compute_total_amount', required=True)
+    expense_accrual_account = fields.Many2one('account.account', readonly=False,
+        domain="[('company_id', '=', company_id), ('internal_type', 'not in', ('receivable', 'payable'))]",
+        related="company_id.expense_accrual_account_id")
+    revenue_accrual_account = fields.Many2one('account.account', readonly=False,
+        domain="[('company_id', '=', company_id), ('internal_type', 'not in', ('receivable', 'payable'))]",
+        related="company_id.revenue_accrual_account_id")
```

Comment:
A helper explaining how those two fields work seems necessary (not the typical stuff: they actually are only used to keep track of the last account used in each type of accrual entry, right ? Seeing it like this without further explanation, one would rather expect them to come from some setting).